### PR TITLE
Enable enterprise cypress tests (#13756)

### PR DESCRIPTION
**Pull request by dacort**
Cherry-picks Cypress Enterprise tests from `master` branch.

### CODE REVIEW

**Review by camsaul — APPROVED** [review]
thanks!

**Comment by nemanjaglumac** [comment]
Since this is going to introduce EE tests to the `release-x.37.x` branch, we should cherry pick two fixes:
1. #13841 already merged as 71c38b2035e98f14b39645080ff76b7f209d4f6e and
2. 13843 (waiting for CI to finish before a merge)

In short, a few tests were failing because they depend on a context `oss` vs `ee`. Those two PRs fix that.

**Pull request merged**
